x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AddEditFlyout/saveConfig.ts

```diff
@@ -62,24 +60,14 @@ export async function saveConfig({
       settings
     };
 
-    if (configurationId) {
-      await callApmApi({
-        pathname: '/api/apm/settings/agent-configuration/{configurationId}',
-        method: 'PUT',
-        params: {
-          path: { configurationId },
-          body: configuration
-        }
-      });
-    } else {
-      await callApmApi({
-        pathname: '/api/apm/settings/agent-configuration/new',
-        method: 'POST',
-        params: {
-          body: configuration
-        }
-      });
-    }
+    await callApmApi({
+      pathname: '/api/apm/settings/agent-configuration',
+      method: 'PUT',
+      params: {
+        query: { overwrite: true },
```

Comment:
Hmm.. good point. So only overwrite when updating - not for creating?

Reply:
It would be a nice safeguard I think. But not absolutely necessary. I was just thinking about the user accidentally overwriting changes from another user, something like that. But that could happen with updating as well.

Reply:
I've added a deterministic id (serialized representation of service.name/environment) and removed the `overwrite` entirely.

Reply:
Damn! I made this change only to realize that we can't do it. It'll require a migration of existing configurations. Bummer, since it simplified it a lot :(